app/controllers/orders_controller.rb

```diff
@@ -1,48 +1,75 @@
 class OrdersController < ApplicationController
-  before_action :find_order, only: [:cart, :checkout, :update, :receipt]
-  before_action :check_access, only: [:cart, :checkout, :receipt]
-
+  before_action :set_order, only: [:cart, :checkout, :add_to_cart, :update, :receipt]
```

Comment:
We know that it's more efficient, but we wanted to avoid bugs down the road if someone added a new action to the controller and didn't see this.